hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/TestUnderReplicatedProcessor.java

```diff
@@ -93,4 +96,40 @@ public void testMessageRequeuedOnException() throws IOException {
     assertEquals(1, queue.underReplicatedQueueSize());
     assertSame(result, queue.dequeueUnderReplicatedContainer());
   }
+
+  @Test
+  public void testMessageNotProcessedIfGlobalLimitReached() throws IOException {
+    AtomicLong inFlightReplications = new AtomicLong(11);
```

Comment:
Can we have a case to check if replication manager is processing all the requests if inflight replication limit factor is set to 0.

Reply:
At line 64 in the diff, I have set the default case to be the "no limit", so if existing tests are working, then they must be OK with that. Adding any more would just be a duplicate of the existing "success" test which runs a container through successfully.